src/json-decoder.ts

```diff
@@ -472,6 +472,44 @@ export namespace JsonDecoder {
     });
   };
 
+  /**
+   * Decoder for a tuple of a specific shape.
+   *
+   * @param decoders The decoders for each element of the tuple.
+   */
+  // This turns a tuple of decoders into a tuple of their results.
+  type TupleOfResults<T extends readonly [] | readonly Decoder<any>[]> = {
+    [K in keyof T]: T[K] extends Decoder<infer R> ? R : never
+  };
+  export const tuple = <T extends readonly [] | readonly Decoder<any>[]>(
+    decoders: T,
+    decoderName: string
+  ): Decoder<TupleOfResults<T>> => {
+    return new Decoder<TupleOfResults<T>>(json => {
+      if (json instanceof Array) {
+        const arr = [];
+        if (json.length !== decoders.length) {
+          return err<TupleOfResults<T>>($JsonDecoderErrors.tupleLengthMismatchError(
+            decoderName, json, decoders));
+        }
+        for (let i =-0; i < json.length; i++) {
```

Comment:
What is this `-0` for?

Reply:
Just a consequence of typing on my laptop keyboard rather than my usual setup. I just sent a new commit to fix it.

Reply:
Ok, I thought that maybe this was some kind of weird performance optimization technique I was not aware of. This things exists! :)
I'll merge release this afternoon and I'll let you know when I'm done.
Thanks!